### PR TITLE
Remove JSON parser from RbxUtility

### PR DESCRIPTION
**Pull request by Fraktality**
Replace the 500-line JSON parser with direct calls to the HttpService methods.

### CODE REVIEW

**Comment by DarraghGriffin** [comment]
Overall I think this is a good change to make because it removes a library we don't need but there is no guarantee that the HttpService methods have the exact same behavior as the current JSON parser in RbxUtility.

I don't think we should remove this without a flag because it could break games in unexpected ways.

**Comment by Fraktality** [comment]
Turns out that there's a functional difference between the RbxUtility and HttpService parsers. RbxUtility handles embedded zeros correctly (encodes as \u0000), but HttpService treats them as terminators.

There are probably people who rely on RbxUtility not being buggy, so I'll close this for now until the HttpService issues are resolved.

**Comment by Fraktality** [comment]
The HttpService issues have been fixed.

**Comment by DarraghGriffin** [comment]
I think we should make this change, but not without having a flag to disable the change if there are unintended effects.

**Comment by Fraktality** [comment]
Sounds good, let me know when you have the flag's name so I can add it to the PR.

**Comment by DarraghGriffin** [comment]
Alright, I will add the flag "RbxUtilityRemoveJsonLibrary" if you add this to the pull request.

**Comment by Fraktality** [comment]
While we're at it, do you think there's still any reason to keep the UserRbxUtilityCreateSetParentLast flag around?

**Comment by kyle-emmerich** [comment]
Nah, I think we can remove that old flag safely.

File: CoreScriptsRoot/Libraries/RbxUtility.lua [hunk 1]
```diff
@@ -11,7 +15,500 @@ local rbxUtilitySetParentLast = (rbxUtilitySetParentLastFlagSuccess == true and
 ------------------------------------------------------------------------------------------------------------------------
 ------------------------------------------------------------------------------------------------------------------------
 
-do
+
+if rbxUtilityRemoveJsonLibrary then
```
Comment (DarraghGriffin) [inline CoreScriptsRoot/Libraries/RbxUtility.lua]: Should be "if not rbxUtilityRemoveJsonLibrary then"

**Comment by hockeychaos** [comment]
will remove this soon, need to resolve some merges

**Pull request closed without merging**